lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1856,6 +1956,20 @@ LogicalResult FIRRTLLowering::visitExpr(SpecialConstantOp op) {
                      getOrCreateIntConstant(APInt(/*bitWidth*/ 1, op.value())));
 }
 
+LogicalResult FIRRTLLowering::visitExpr(SubindexOp op) {
+  if (isZeroBitFIRRTLType(op->getResult(0).getType()))
+    return setLowering(op, Value());
+
+  auto resultType = lowerType(op->getResult(0).getType());
+  Value value = getPossiblyInoutLoweredValue(op.input());
+  assert(value && "subindex lowering failed");
```

Comment:
Is there a better message that we can generate here other than an assert and "lowering failed"? 😅

Reply:
Exactly. I changed to emit an error